File: src/main/java/com/raoulvdberge/refinedstorage/api/util/StackListEntry.java
package com.raoulvdberge.refinedstorage.api.util;

import java.util.UUID;

/**
 * Represents a stack in a stack list.
 *
 * @param <T> the stack type
 */
public class StackListEntry<T> {
    private UUID id;
    private T stack;

    public StackListEntry(T stack) {
        this.stack = stack;
        this.id = UUID.randomUUID();
    }

    public StackListEntry(UUID id, T stack) {
        this.id = id;
        this.stack = stack;
    }

    /**
     * The unique id of the entry.
     * This id is NOT persisted, nor does it hold any relation to the contained stack.
     * It is randomly generated.
     *
     * @return the id
     */
    public UUID getId() {
        return id;
    }

    /**
     * @return the stack
     */
    public T getStack() {
        return stack;
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/util/StackListEntry.kt
package com.refinedmods.refinedstorage.api.util

import java.util.*


/**
 * Represents a stack in a stack list.
 *
 * @param <T> the stack type
</T> */
class StackListEntry<T> {
    /**
     * The unique id of the entry.
     * This id is NOT persisted, nor does it hold any relation to the contained stack.
     * It is randomly generated.
     *
     * @return the id
     */
    val id: UUID

    /**
     * @return the stack
     */
    val stack: T

    constructor(stack: T) {
        this.stack = stack
        id = UUID.randomUUID()
    }

    constructor(id: UUID, stack: T) {
        this.id = id
        this.stack = stack
    }
}